src/beacon/app/device_performance_endpoint.py

```
@@ -13,202 +13,389 @@
+            performance_data['recommendations'] = generate_maintenance_recommendations(
+                device_info, performance_data['summary'], readings_data
+            )
         
         return create_json_response(performance_data)
         
     except HTTPException:
         raise
     except Exception as e:
-        print(f"Error in get_device_performance: {str(e)}")
+        print(f"Error in get_comprehensive_device_performance: {str(e)}")
         raise HTTPException(status_code=500, detail=f"Failed to fetch performance metrics: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)